x/clp/keeper/table_test.go

```diff
@@ -4,16 +4,51 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
-	"testing"
-
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"testing"
 )
 
+// THe normalization has been shifted to a different module , so these tests cannot remain pure anymore , unless we add normalization factor to the json file directly
+// Using a stub now for providing hardcoded values of normalization-factor and adjustment flag
+// The stub implementation can be replaced with DenomWhitelist types
+
+func GetNormalizationMap() map[string]int64 {
+	m := make(map[string]int64)
+	m["cel"] = 4
+	m["ausdc"] = 6
+	m["usdt"] = 6
+	m["usdc"] = 6
+	m["cro"] = 8
+	m["cdai"] = 8
+	m["wbtc"] = 8
+	m["ceth"] = 8
+	m["renbtc"] = 8
+	m["cusdc"] = 8
+	m["husd"] = 8
+	m["ampl"] = 9
+	return m
+}
+func GetNormalizationFactorStub(denom string) (sdk.Dec, bool) {
```

Comment:
We should test the actual implementation of this that exists in the clp keeper.

The test should insert register the denoms in the whitelist keeper so it can be fetched from there.

Can use a mock generator for whitelist keeper if needed: https://github.com/golang/mock

Reply:
That can be a different test , this test is pure. It just tests the mathematical  formulae, this file does not have anything related to the blockchain / state . 
We have a  counterpart for these on the frontend as well.